packages/keplr/src/wallet-connect/main-wallet.ts

```diff
@@ -0,0 +1,109 @@
+import { ChainName, ChainRegistry, Dispatch, State } from '@cosmos-kit/core';
+import { MainWalletBase } from '@cosmos-kit/core';
+import { KeplrWalletConnectV1 } from '@keplr-wallet/wc-client';
+import WalletConnect from '@walletconnect/client';
+import EventEmitter from 'events';
+
+import { ChainWCKeplr } from './chain-wallet';
+import { WCKeplrData } from './types';
+
+export class WCKeplrWallet extends MainWalletBase<
+  KeplrWalletConnectV1,
+  WCKeplrData,
+  ChainWCKeplr
+> {
+  protected _chains: Map<ChainName, ChainWCKeplr>;
+  protected _client: KeplrWalletConnectV1;
+  connector: WalletConnect;
+  ee: EventEmitter;
+
+  constructor(_concurrency?: number) {
+    super(_concurrency);
+
+    this.connector = new WalletConnect({
+      bridge: 'https://bridge.walletconnect.org'
+    });
+
+    this.connector.on("connect", (error) => {
+      if (error) {
+        throw error;
+      }
+      this.setClient(new KeplrWalletConnectV1(this.connector));
+      this.ee.emit('update');
+    });
+
+    this.connector.on("disconnect", (error) => {
+      if (error) {
+        throw error;
+      }
+      this.ee.emit('disconnect');
+    });
+
+    this._client = new KeplrWalletConnectV1(this.connector);
+    this.ee = new EventEmitter();
+  }
+
+  get isInSession() {
+    return this.connector.connected;
+  }
+
+  get qrUri() {
+    return this.connector.uri;
+  }
+
+  get client() {
+    return this._client;
+  }
+
+  setClient(client: KeplrWalletConnectV1) {
+    this._client = client;
+  }
+
+  protected setChains(supportedChains: ChainRegistry[]): void {
```

Comment:
this should be `Chain` from `@chain-registry/types`

Reply:
for now it is ChainInfo[]

```
export interface ChainInfo extends Info<ChainName> {
  registry?: Chain;
  options?: {
    stargate?: (chainInfo: Chain) => SigningStargateClientOptions | undefined;
    cosmwasm?: (chainInfo: Chain) => SigningCosmWasmClientOptions | undefined;
  }
}
```